android-components/components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonsManagerAdapterTest.kt

```diff
@@ -552,11 +566,46 @@ class AddonsManagerAdapterTest {
         verify(addonsManagerAdapterDelegate).onFindMoreAddonsButtonClicked()
     }
 
+    @Test
+    fun bindHeaderButton() {
+        val store = BrowserStore(initialState = BrowserState(extensionsProcessDisabled = true))
+        val adapter = spy(AddonsManagerAdapter(mock(), mock(), emptyList(), mock(), emptyList(), store))
+
+        val restartButton = TextView(testContext)
+        val viewHolder = CustomViewHolder.HeaderViewHolder(View(testContext), restartButton)
+        adapter.bindHeaderButton(viewHolder)
+        assertEquals(1, adapter.currentList.size)
+
+        viewHolder.restartButton.performClick()
+        dispatcher.scheduler.advanceUntilIdle()
+
+        assertFalse(store.state.extensionsProcessDisabled)
+        verify(adapter).submitList(emptyList())
+    }
+
+    @Test
+    fun testNotificationShownWhenProcessIsDisabled() {
+        val store = BrowserStore(initialState = BrowserState(extensionsProcessDisabled = true))
+        val adapter = AddonsManagerAdapter(mock(), mock(), emptyList(), mock(), emptyList(), store)
+
+        val itemsWithSections = adapter.createListWithSections(emptyList())
+        assertEquals(AddonsManagerAdapter.HeaderSection, itemsWithSections.first())
+    }
+
+    @Test
+    fun testNoNotificationShownWhenProcessIsEnabled() {
```

Comment:
nit: `testNoNotificationShownWhenProcessIsEnabled` might be more clear as `testNotificationNotShownWhenProcessIsEnabled`